scripts/runtests.sh

```diff
@@ -254,6 +254,28 @@ python3 scripts/check-yaml-fields.py "$tmpdir"/var/log/installer/autoinstall-use
         'autoinstall.source.id="ubuntu-server-minimal"'
 grep -q 'finish: subiquity/Install/install/postinstall/run_unattended_upgrades: SUCCESS: downloading and installing security updates' $tmpdir/subiquity-server-debug.log
 
+clean
+LANG=C.UTF-8 timeout --foreground 60 \
+    python3 -m subiquity.cmd.tui \
+    --dry-run \
+    --output-base "$tmpdir" \
+    --machine-config examples/simple.json \
+    --autoinstall examples/autoinstall-ad.yaml \
+    --kernel-cmdline autoinstall \
+    --source-catalog examples/mixed-sources.yaml &
+sub_pid=$!
+until [[ -S "$tmpdir/socket" || $(ps -p "$sub_pid" > /dev/null ) == 1 ]]
+do
+    sleep 1
+done
+curl --unix-socket "$tmpdir/socket" -X POST 'http://localhost/meta/client_variant?variant=%22desktop%22'
+ad_info=$(curl --unix-socket "$tmpdir/socket" -X GET 'http://localhost/active_directory' | jq '.password="PassWord"')
+# User and domain name (but not passwd) should come from the autoinstall file.
+curl --unix-socket "$tmpdir/socket" -X POST -d "${ad_info}" 'http://localhost/active_directory'
```

Comment:
This is what our API tests looked like at first.  I suspect driving this test from there is going to be simpler.  I want this PR done but I want to see you follow up with this moved to an API test.

Reply:
or move it there now if that's simpler

Reply:
Simpler to move it now, indeed. That results in code much better to reason about. The only gotcha is that I needed to expose the `output_base` dir in the testing Server class, so I could inspect the logs as part of the test case.

Reply:
Thank you.  And no problem on exposing the output_base value, that's a reasonable change for the test you had in mind.